src/utp_connections.rs

```diff
@@ -65,59 +60,19 @@ mod test {
     use std::time::Duration;
     use sender_receiver::CrustMsg;
 
-    fn listen(port: u16) -> io::Result<UtpListener> {
-        UtpListener::bind(("0.0.0.0", port))
-    }
-
     #[test]
-    fn cannot_establish_connection() {
-        let listener = UdpSocket::bind({
-                           net::SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 0)
-                       })
-                           .unwrap();
-        let port = listener.local_addr().unwrap().port();
-        drop(listener);
-        let _err =
-            connect_utp(SocketAddr(net::SocketAddr::V4(net::SocketAddrV4::new(Ipv4Addr::new(127,
-                                                                                       0,
-                                                                                       0,
-                                                                                       1),
-                                                                         port))))
-                .err()
-                .unwrap();
-    }
+    fn send_receive_data() {
+        let socket_0 = unwrap_result!(UdpSocket::bind("0.0.0.0:0"));
+        let socket_1 = unwrap_result!(UdpSocket::bind("0.0.0.0:0"));
 
-    #[test]
-    fn establishing_connection() {
-        let listener = listen(0).unwrap();
-        let port = listener.local_addr().unwrap().port();
-
-        let handle = spawn(move || listener.accept().unwrap());
-
-        // Note: when the result of connect_utp here is assigned to a variable
-        // named _, this test takes much longet to complete. My guess is that
-        // it happens because _ is dropped immediately, but any other named
-        // variable is dropped only at the end of the scope. So when naming
-        // this variable, the socket outlives the above thread, which somehow
-        // makes this test finish faster for some reason.
-        let _socket = connect_utp(SocketAddr(net::SocketAddr::V4(net::SocketAddrV4::new(Ipv4Addr::new(127,
-                                                                                                      0,
-                                                                                                      0,
-                                                                                                      1),
-                                                                                        port))))
-                    .unwrap();
-
-        let _ = handle.join().unwrap();
-    }
+        let port_0 = unwrap_result!(socket_0.local_addr()).port();
+        let port_1 = unwrap_result!(socket_1.local_addr()).port();
 
-    #[test]
-    fn send_receive_data() {
-        let listener = listen(0).unwrap();
-        let port = listener.local_addr().unwrap().port();
+        let addr_0 = SocketAddr(net::SocketAddr::V4(net::SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), port_0)));
```

Comment:
It would make more sense to add a new function like

```
fn SocketAddr::localhost_v4(port: u16) -> SocketAddr;
```

Actually I think that should be added to the standard library. I might raise a rust-lang/rfcs bug about it.